packages/core/upload/admin/src/components/SelectTree/utils/flattenTree.ts

```diff
@@ -0,0 +1,26 @@
+type TreeNode<T> = {
+  value: T;
+  children?: TreeNode<T>[];
+  label?: string;
+};
+
+export type FlattenedNode<T> = {
+  value: T;
+  parent?: T;
+  depth: number;
+  // we need the label in places where flattenTree is used
+  label?: string;
+  children?: TreeNode<T>[];
+};
+
+export default function flattenTree<T>(
+  tree: TreeNode<T>[],
+  parent: TreeNode<T> | null = null,
+  depth: number = 0
+): FlattenedNode<T>[] {
+  return tree.flatMap((item) =>
```

Comment:
just in case, is it ok that the parent can be null in the tree, but undefined in the flattened node?

Reply:
oh nvm, just saw you renamed the file to ts :D

Reply:
Ok